internal/controller/pkg/revision/watch.go

```diff
@@ -0,0 +1,87 @@
+/*
```

Comment:
Thank you @negz for basically writing all of this code for me :)

(copied with slight modifications from `EnqueueRequestForAllRevisionsWithRequests`)